var/spack/repos/builtin/packages/pexsi/package.py

```diff
@@ -95,7 +95,7 @@ def edit(self, spec, prefix):
             'make.inc'
         )
         shutil.copy(template, makefile)
-        for key, value in substitutions.items():
+        for key, value in sorted(substitutions.items(), reverse=True):
```

Comment:
Sorting this by key happens to work here, but that's only by coincidence, it doesn't give us much more control than alphabetic ordering. Let's replace the dictionary with a list of 2-tuples. That way we have explicit control over the ordering. This is how the `hpl` package works.

Reply:
Thank you for your comment.
I changed to use a list of tuples.